perception/autoware_tensorrt_bevdet/include/autoware/tensorrt_bevdet/bevpool_plugin.hpp

```diff
@@ -0,0 +1,123 @@
+  ~BEVPoolPlugin();
+
+  // Method inherited from IPluginV2
+  const char * getPluginType() const noexcept override;
+  const char * getPluginVersion() const noexcept override;
+  int32_t getNbOutputs() const noexcept override;
+  int32_t initialize() noexcept override;
+  void terminate() noexcept override;
+  size_t getSerializationSize() const noexcept override;
+  void serialize(void * buffer) const noexcept override;
+  void destroy() noexcept override;
+  void setPluginNamespace(const char * pluginNamespace) noexcept override;
+  const char * getPluginNamespace() const noexcept override;
+
+  // Method inherited from IPluginV2Ext
+  DataType getOutputDataType(
+    int32_t index, DataType const * inputTypes, int32_t nbInputs) const noexcept override;
+  void attachToContext(
+    cudnnContext * contextCudnn, cublasContext * contextCublas,
+    IGpuAllocator * gpuAllocator) noexcept override;
+  void detachFromContext() noexcept override;
+
+  // Method inherited from IPluginV2DynamicExt
+  IPluginV2DynamicExt * clone() const noexcept override;
+  DimsExprs getOutputDimensions(
+    int32_t outputIndex, const DimsExprs * inputs, int32_t nbInputs,
+    IExprBuilder & exprBuilder) noexcept override;
+  bool supportsFormatCombination(
+    int32_t pos, const PluginTensorDesc * inOut, int32_t nbInputs,
+    int32_t nbOutputs) noexcept override;
+  void configurePlugin(
+    const DynamicPluginTensorDesc * in, int32_t nbInputs, const DynamicPluginTensorDesc * out,
+    int32_t nbOutputs) noexcept override;
+  size_t getWorkspaceSize(
+    const PluginTensorDesc * inputs, int32_t nbInputs, const PluginTensorDesc * outputs,
+    int32_t nbOutputs) const noexcept override;
+  int32_t enqueue(
+    const PluginTensorDesc * inputDesc, const PluginTensorDesc * outputDesc,
+    const void * const * inputs, void * const * outputs, void * workspace,
+    cudaStream_t stream) noexcept override;
+
+protected:
+  // To prevent compiler warnings
+  using nvinfer1::IPluginV2::enqueue;
+  using nvinfer1::IPluginV2::getOutputDimensions;
+  using nvinfer1::IPluginV2::getWorkspaceSize;
+  using nvinfer1::IPluginV2Ext::configurePlugin;
+};
+
+class BEVPoolPluginCreator : public IPluginCreator
+{
+private:
+  static PluginFieldCollection fc_;
+  static std::vector<PluginField> attr_;
+  std::string namespace_;
+
+public:
+  BEVPoolPluginCreator();
+  ~BEVPoolPluginCreator();
+  const char * getPluginName() const noexcept override;
+  const char * getPluginVersion() const noexcept override;
+  const PluginFieldCollection * getFieldNames() noexcept override;
+  IPluginV2DynamicExt * createPlugin(
+    const char * name, const PluginFieldCollection * fc) noexcept override;
+  IPluginV2DynamicExt * deserializePlugin(
+    const char * name, const void * serialData, size_t serialLength) noexcept override;
+  void setPluginNamespace(const char * pluginNamespace) noexcept override;
+  const char * getPluginNamespace() const noexcept override;
+};
+
```

Comment:
The closing namespace comment `}  // namespace nvinfer1` should be aligned with the opening namespace statement for better readability. Consider using a consistent style throughout the file.